### PR TITLE
Introduce 'exclude_tags'

### LINKED ISSUE #118: Add skip-tags functionality
This wouldn't change the structure of the `openshift_cluster_content`.

But would allow for being able to skip over certain sections that we didn't want to be applied using the existing `tags` functionality

envisioning something very similar to what ansible already does, but passed as an extra var `-e skip_tags=jenkins-persistent,nexus` 

name of the variable might be confusing with native ansible `--skip-tags` so maybe something like `applier_skip_tags` might be better

### PR DESCRIPTION
**Pull request by bvkin**
#### What does this PR do?
Adds the skip_tags option to be used in the same manner as filter_tags.

#### How should this be tested?
Specify the skip_tags variable when running openshift_applier against an inventory which has tags set, verify that no objects with the given tag were applied. You can also verify that the filter_tags option still works along with neither option. 

#### Is there a relevant Issue open for this?
resolves #118 

#### Who would you like to review this?
cc: @redhat-cop/openshift-applier

### CODE REVIEW

**Comment by pcarney8** [comment]
+1 to both things that @oybed said, also i created #127 to change that variable name from `filter_tags` to `include_tags` :smile:

File: roles/openshift-applier/README.md [hunk 1]
```diff
@@ -192,12 +192,15 @@ metadata:
 
 ### Filtering content based on tags
 
-The `openshift-applier` supports the use of tags in the inventory (see example above) to allow for filtering which content should be processed and not. The `filter_tags` variable/fact takes a comma separated list of tags that will be processed and only content with matching tags will be applied.
+The `openshift-applier` supports the use of tags in the inventory (see example above) to allow for filtering which content should be processed and not. The `filter_tags` variable/fact takes a comma separated list of tags that will be processed and only content with matching tags will be applied. Additionally, tags can be skipped in the same manner with the `exclude_tags` option. 
+
+**_NOTE:_** These two options are mutually exclusive.
```
Comment (oybed) [inline roles/openshift-applier/README.md]: Don't think this is true anymore ... + probably should include some info re: not allowing the same tags as part of both options

File: roles/openshift-applier/README.md [hunk 2]
```diff
@@ -192,12 +192,15 @@ metadata:
...
 
 **_NOTE:_** Entries in the inventory without tags will not be processed when a valid list is supplied with the `filter_tags` option.
 
 ```
 filter_tags=tag1,tag2
+exclude_tags=tag1,tag2
```
Comment (oybed) [inline roles/openshift-applier/README.md]: This is example violates the "rules" as the same tags should not be allowed in both - i.e.: this should be an error case. Please update `exclude_tags` to be `tag3,tag4`.

File: roles/openshift-applier/README.md (hunk 1 above)
Comment (oybed) [reply roles/openshift-applier/README.md]: Also need to include some info on which one has precedence  - i.e.: if both `filter` (future `include`) and `exclude` are used and an entry has a list of tags whereas one exists in the `include` and another in `exclude` .

**Comment by oybed** [comment]
Also, please fix the error found by the CI job:
```
--> Executing Yamllint on files found in /home/travis/build/redhat-cop/openshift-applier/...
    /home/travis/build/redhat-cop/openshift-applier/roles/openshift-applier/tasks/error-on-unsupported.yml
      26:39     error    no new line character at the end of file  (new-line-at-end-of-file)
```

**Comment by etsauer** [comment]
@oybed where are we with this?

**Comment by oybed** [comment]
@etsauer I'll sync with @bvkin to check status and get it done

**Comment by bvkin** [comment]
@oybed @etsauer I think it's done, but could probably use another review as I made a lot of changes to handle overlapping `include_tags`  and `exclude_tags`.

**Comment by oybed** [comment]
Yes, let's do a (final) review today when we get a chance to sync. Noticed a few things that still need some work, but we should be able to get it done quickly.

**Comment by oybed** [comment]
CI passed - merging.

**Pull request merged**